version: '2.2'
services:
    users_actions_app:
      build:
        context: .
      ports:
        - "5000:5000"
      volumes:
        - ./logs/users_actions_app:/users_actions_app/logs
      command: gunicorn users_actions_app.wsgi_app:app --bind 0.0.0.0:5000  --reload
      networks:
        - moves_network

    nginx:
      image: nginx:1.19.2
      volumes:
        - /tmp/logs/nginx/:/var/log/nginx/
        - ./deploy/etc/nginx/nginx.conf:/etc/nginx/nginx.conf:ro
        - ./deploy/etc/nginx/conf.d:/etc/nginx/conf.d:ro

      depends_on:
        - users_actions_app
      ports:
        - 8080:80
      networks:
        - moves_network

    filebeat:
      image: docker.elastic.co/beats/filebeat:8.7.0-arm64
      volumes:
        - /var/lib/docker/containers:/var/lib/docker/containers:ro
        - /tmp/logs/nginx:/var/log/nginx:ro
        - ./logs/users_actions_app:/users_actions_app/logs
        - ./deploy/filebeat.yml:/usr/share/filebeat/filebeat.yml
      depends_on:
        - users_actions_app
        - nginx
      links:
        - logstash
      networks:
        - moves_network

networks:
  moves_network:
    name: moves_network
    external: true